Add gradlePluginPortal(Action<>) to RepositoryHandler

StefMa opened the pull request:
### Context
This is a "follow up" of #8636 
I think @ljacomet has just overseen the `gradlePluginPortal` 😅.

I hope it is ok and break nothing to change the PluginPortal from a `ArtifactRepository` to an  `MavenArtifactRepository`. But since it is an "up cast" nothing should break ✊.

I also have added the `@Incubating` because the `gradlePluginPortal()` have this as well...

Changes in the `accepted-public-api-changes.json` (like [here](https://github.com/gradle/gradle/pull/8636/files#diff-d47ed06f7d6d79e3f3c3fc7268b7939e)) and in the "changelog" ([here](https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/release/notes.md)) should be done afterwards as well 👍 

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] ~Update User Guide, DSL Reference, and Javadoc for public-facing changes~
- [ ] ~Ensure that tests pass locally: `./gradlew <changed-subproject>:check`~

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

ljacomet commented on subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/RepositoryHandler.java:
```diff
@@ -88,7 +88,17 @@
      */
     @Incubating
-    ArtifactRepository gradlePluginPortal();
+    MavenArtifactRepository gradlePluginPortal();
```
I do not think there is a need to make this type more specific and sadly it is a binary breaking change. So please revert it back to the `ArtifactRepository`

ljacomet commented on subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/RepositoryHandler.java:
```diff
@@ -88,7 +88,17 @@
+     * @since 5.3
+     */
+    @Incubating
+    MavenArtifactRepository gradlePluginPortal(Action<? super MavenArtifactRepository> action);
```
For consistency, and because having the support for `mavenContent` really does not mean a lot in this case, I would use `ArtifactRepository` here.

ljacomet commented on subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/RepositoryHandler.java:
```diff
@@ -88,7 +88,17 @@
+     * 
+     * @param action a configuration action
+     * @return the added resolver
+     * @since 5.3
```
This needs to say `@since 5.4` and it will make the API change go away.

StefMa commented:
@ljacomet fixed and CI is 🍏 🤘

ljacomet commented:
@StefMa Would you consider squashing your commits? Not sure the history is useful given the scope of the change. Wdyt?

Then I would trigger a full build on our CI infra to make sure all is good.

StefMa commented:
Sure I could do this.
But you can do this while merging as well ☺
See
https://help.github.com/en/articles/merging-a-pull-request#merging-a-pull-request-on-github

On Mon, Mar 18, 2019, 4:52 PM Louis Jacomet <notifications@github.com>
wrote:

> @StefMa <https://github.com/StefMa> Would you consider squashing your
> commits? Not sure the history is useful given the scope of the change. Wdyt?
>
> Then I would trigger a full build on our CI infra to make sure all is good.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gradle/gradle/pull/8802#issuecomment-473971500>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJwYewWJ9I-w1PFr4HpMexDRLTT2bLbpks5vX7YmgaJpZM4b51GG>
> .
>

ljacomet commented:
That's true, though I have no idea how it plays with our DCO check 😉

ljacomet commented:
Let me trigger a full build now.

StefMa commented:
Rebased and squashed everything ✅ 

This https://builds.gradle.org/viewLog.html?buildId=20995831&buildTypeId=Gradle_Check_Stage_ReadyforMerge_Trigger was failing on your CI... 
For whatever reasons I can't access it currently...
Maybe you can check that out?

StefMa commented:
How I can fix this issues? 🤔

ljacomet commented:
My bad, forgot to leave a comment last week. The branch point you took was bad in the sense that it had a performance regression. It has been fixed since then.
A rebase of the branch on current master is your best course of action.

Thanks for your patience.

StefMa commented:
Rebased ✅

On Mon, Mar 25, 2019, 3:16 PM Louis Jacomet <notifications@github.com>
wrote:

> My bad, forgot to leave a comment last week. The branch point you took was
> bad in the sense that it had a performance regression. It has been fixed
> since then.
> A rebase of the branch on current master is your best course of action.
>
> Thanks for your patience.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gradle/gradle/pull/8802#issuecomment-476215889>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJwYe7s2FT-73QXjwzfIvvogdxqK7rJsks5vaNojgaJpZM4b51GG>
> .
>

ljacomet commented:
And merged, thanks for the contribution
